File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        answerList = ListNode()
        currentN = answerList

        while (list1 and list2):
            if (list1.val <= list2.val):
                currentN.next = list1
                list1 = list1.next
            else:
                currentN.next = list2
                list2 = list2.next
            currentN = currentN.next

        if (list1):
            currentN.next = list1
        else:
            currentN.next = list2

        return answerList.next        

File: Data Structures & Algorithms/add-two-numbers/submission-1.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} l1
     * @param {ListNode} l2
     * @return {ListNode}
     */
    addTwoNumbers(l1, l2) {
        let dummy = new ListNode();
        let cur = dummy;

        let carry = 0;
        while (l1 || l2 || carry) {
            let val1 = l1 ? l1.val : 0;
            let val2 = l2 ? l2.val : 0;

            let sum = val1 + val2 + carry;
            carry = Math.floor(sum/10);
            sum = sum % 10;

            cur.next = new ListNode(sum);

            cur = cur.next;
            l1 = l1 ? l1.next : null;
            l2 = l2 ? l2.next : null;
        }
        return dummy.next;
    }
}
